icu4j/main/classes/core/src/com/ibm/icu/util/ULocale.java

```diff
@@ -1403,10 +1403,15 @@ private static synchronized void loadAliasData() {
                         "Incorrect key [" + aliasFrom + "] in alias:territory.");
                 }
                 if (aliasTo.length() < 3 || aliasTo.length() > 8) {
-                    // Ignore replacement < 3 for now. see CLDR-14312
-                    // throw new IllegalArgumentException(
-                    //    "Incorrect value [" + aliasTo + "] in alias:subdivision.");
-                    continue;
+                    if (aliasTo.length() == 2) {
```

Comment:
It looks like in C++ you keep the alias data as is and add the zzzz while canonicalizing -- but here in Java you modify the alias data.
Why the difference?
Is it possible that subdivision aliases may be used in some place where the added zzzz would not be desirable? In that case, the C++ approach seems better.

Reply:
In C++, the alias data is directly referencing to the memory in the memory mapped resources, so we cannot grow the memory for 4 other char. In Java, it is separately allocated.